README.md

```diff
@@ -89,19 +94,96 @@ int main()
+
+    Azure::Response<Models::UploadBlockBlobResult> response = block.Upload(stream);
+
+    Models::UploadBlockBlobResult model = response.Value;
+    printf("Last modified date of uploaded blob: %s\n", model.LastModified.ToString().c_str());
   }
   catch (const Azure::Core::RequestFailedException& e)
   {
+    printf("Status Code: %d, Reason Phrase: %s\n", (int)e.StatusCode, e.ReasonPhrase.c_str());
     std::cout << e.what() << std::endl;
     return 1;
   }
   return 0;
 }
 ```
 
-Understanding the key concepts from the `azure-core-cpp` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use. You can find more information about them, with sample code snippets, here: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core#key-concepts
+#### Key Core concepts
+
+Understanding the key concepts from the `azure-core-cpp` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use.
+
+The main shared concepts of `Azure::Core` include:
+
+- Accessing HTTP response details for the returned model of any SDK client operation, via `Response<T>`.
+- Exceptions for reporting errors from service requests in a consistent fashion via the base exception type `RequestFailedException`.
+- Abstractions for Azure SDK credentials (`TokenCredential`).
+- Handling streaming data and input/output (I/O) via `BodyStream` along with its derived types.
+- Polling long-running operations (LROs), via `Operation<T>`.
+- HTTP pipeline and HTTP policies such as retry and logging, which are configurable via service client specific options.
+- Replaceable HTTP transport layer to send requests and receive responses over the network.
+
+#### `Response <T>` Model Types
+
+Many client library operations **return** the templated `Azure::Core::Response<T>` type from the API calls. This type let's you get the raw HTTP response from the service request call the Azure service APIs make, along with the result of the operation to get more API specific details. This is the templated `T` operation result which can be extracted from the response, using the `Value` field.
+
+```C++
+  std::string containerName = "testcontainer";
+
+  BlobContainerClient containerClient
+      = BlobContainerClient::CreateFromConnectionString(GetConnectionString(), containerName);
+
+  // Azure service operations return a Response<T> templated type
+  Azure::Response<Models::CreateBlobContainerResult> createResult
+      = containerClient.CreateIfNotExists();
+
+  // You can extract the T, from the returned Response<T>,
+  // which is typically named with a Result suffix in the type name.
+  Models::CreateBlobContainerResult result = createResult.Value;
+
+  // Now you can look at API specific members on the result object that is returned.
+  printf("Did the container get created? %d\n", result.Created);

```

Comment:
std::cout

Reply:
Do you know what's the best way to emit a bool Anton?
Can you please share a suggestion/code snippet.

Reply:
`std::cout << "Did the container get created? " << std::boolalpha << result.Created << std::endl;`?
https://onlinegdb.com/SFgGxu_S-7

But I would not bother to put it, if C++ don't have it by default, plus applying `std::boolalpha` changes the behavior in all other statements in the program. Unless you set and restore ios flags, but all that diverts the attention. I'd say just do "`std::cout << "Did the container get created? " << result.Created << std::endl;`" - it is pretty good to illustrate the SDK without diverting attention.